docs/doc/60-contributing/03-rfcs/20221017-multiple-catalog.md

```diff
@@ -0,0 +1,138 @@
+---
+title: Multiple Catalog
+description: multiple catalog support for databend
+---
+
+- RFC PR: [datafuselabs/databend#8255](https://github.com/datafuselabs/databend/pull/8255)
+- Tracking Issue: [datafuselabs/databend#0000](https://github.com/datafuselabs/databend/issues/0000)
+
+## Summary
+
+Allow users to maintain multiple catalogs for the databend.
+
+## Motivation
+
+Databend organize data in three layers:
+
+```txt
+catalog -> database -> table
+```
+
+- `catalog`: the biggest layer for databend, contains all databases and tables, provided by [`Catalog`](https://github.com/datafuselabs/databend/blob/556aedc00e5e8a95a7551d0ec21b8e6fa7573e0a/src/query/catalog/src/catalog.rs#L80)
+- `database`: the container of tables, provided by [`Database`](https://github.com/datafuselabs/databend/blob/556aedc00e5e8a95a7551d0ec21b8e6fa7573e0a/src/query/catalog/src/database.rs#L44)
+- `table`: the smallest unit in databend, provided by [`Table`](https://github.com/datafuselabs/databend/blob/556aedc00e5e8a95a7551d0ec21b8e6fa7573e0a/src/query/catalog/src/table.rs#L44)
+
+By default, all databases and tables will be stored in `default` catalog (powered by `metasrv`).
+
```

Comment:
Are there other options to store databases and tables besides `metasrv`?

Do the db/tables in various catalog share the same data type?

Reply:
> Are there other options to store databases and tables besides `metasrv`?

- For `hive` catalog, databases and tables will be stored in `hive meta store`, which could be Postgres or MySQL.
- For `iceberg`, databases and tables will be stored in AVRO files.

> Do the db/tables in various catalog share the same data type?

No, the data types for the database and table are catalog implementation detail.